pkg/relay/relay.go

```diff
@@ -287,6 +296,15 @@ func (rs *Relay) GetPayload(ctx context.Context, payloadRequest *types.SignedBli
 
 	// defer put delivered datastore write
 	go func(rs *Relay, slot structs.Slot, trace structs.DeliveredTrace) {
+		if rs.config.PublishBlock {
```

Comment:
I'm not super sure about that, but if we're doing that - I think this should happen after rs.d.putDelivered - not before

Reply:
Why after `PutDelivered`? PutDelivered is only for updating the Data API

Reply:
Because the integrity of dataAPI imho trumps block publishing, but if you like to keep it like that, not a big deal

Reply:
And does not affect to whether the call will be successful or not. In case of error we simply log it

Reply:
So, you suggest not to publish the block it if the `PutDelivered` call fails? I think the Data API should be relatively decoupled from the relay operations.

Reply:
I was thinking about something more like a precaution, like if publish block (randomly?) panics, to be paranoic about data api having it published.

Reply:
Okay sure, paranoid-drive-development is always good ;) It does not have any drawbacks, so I'll move it to after the call